astropy/units/quantity.py

```diff
@@ -714,7 +714,8 @@ def to_value(self, unit=None, equivalencies=[]):
                     # not in-place!
                     value = value * scale
 
-        return value if self.shape else value.item()
+        return value if self.shape else (value[()] if self.dtype.fields
```

Comment:
Probably this should be unconditionally `value[()]` instead of `value.item()` but out of scope of this PR...